### PR TITLE
[Storage] Bug fix - Upload from requests get fails in get_length

### LINKED ISSUE #8500: [Storage Blob] Inspect uploading blobs with no length

### PR DESCRIPTION
**Pull request by rakshith91**
Fixes #8500 

Look at the tests added for context.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/request_handlers.py [hunk 1]
```diff
@@ -66,7 +66,10 @@ def get_length(data):
         except (AttributeError, UnsupportedOperation):
             pass
         else:
-            return fstat(fileno).st_size
+            try:
+                return fstat(fileno).st_size
+            except (OSError):
```
Comment (johanste) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_shared/request_handlers.py]: Why parenthesis?

Comment (rakshith91) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_shared/request_handlers.py]: fixing it :)

File: sdk/storage/azure-storage-blob/tests/recordings/test_common_blob_async.test_create_blob_with_requests_async.yaml [hunk 2]
```diff
@@ -0,0 +1,11811 @@
+interactions:
```
Comment (johanste) [inline sdk/storage/azure-storage-blob/tests/recordings/test_common_blob_async.test_create_blob_with_requests_async.yaml]: Do we really need this large recording? Our repo is bloating waaay to much right now.

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/request_handlers.py [hunk 3]
```diff
@@ -66,7 +66,10 @@ def get_length(data):
         except (AttributeError, UnsupportedOperation):
             pass
         else:
-            return fstat(fileno).st_size
+            try:
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_shared/request_handlers.py]: a comment here could be perfect!!(explaining why we need to try inside of else block)

Comment (rakshith91) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_shared/request_handlers.py]: done

File: sdk/storage/azure-storage-blob/tests/recordings/test_common_blob_async.test_create_blob_with_requests_async.yaml (hunk 2 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-blob/tests/recordings/test_common_blob_async.test_create_blob_with_requests_async.yaml]: Hmm, if we want to run this test :)

File: sdk/storage/azure-storage-blob/tests/test_common_blob.py [hunk 4]
```diff
@@ -247,6 +247,34 @@ def test_create_blob_with_lease_id(self, resource_group, location, storage_accou
         content = blob.download_blob(lease=lease).readall()
         self.assertEqual(content, data)
 
+    @GlobalStorageAccountPreparer()
+    def test_create_blob_with_generator(self, resource_group, location, storage_account, storage_account_key):
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-blob/tests/test_common_blob.py]: I'm wondering if the content size that the generator could produce is larger than chunk size, it is still working.
let's say if the chunk size is 4096, the generator generate "hello" every time, and the generator could generate 820 "hello" in total. Not sure if it is working. I'm also checking

File: sdk/storage/azure-storage-blob/tests/recordings/test_common_blob_async.test_create_blob_with_requests_async.yaml (hunk 2 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-blob/tests/recordings/test_common_blob_async.test_create_blob_with_requests_async.yaml]: just noticed how big they are...removed and skipping for recording

File: sdk/storage/azure-storage-blob/tests/test_common_blob.py (hunk 4 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-blob/tests/test_common_blob.py]: Will be fixed in #8513

File: sdk/storage/azure-storage-blob/tests/test_common_blob_async.py [hunk 5]
```diff
@@ -304,6 +305,50 @@ def _assert_blob_not_soft_deleted(self, blob):
...
+    @GlobalStorageAccountPreparer()
+    @AsyncStorageTestCase.await_prepared_test
+    async def test_create_blob_with_aiohttp_async(self, resource_group, location, storage_account, storage_account_key):
+        await self._setup(storage_account.name, storage_account_key)
+        blob = self.bsc.get_blob_client(self.container_name, "gutenberg")
+        # Act
+        uri = "http://www.gutenberg.org/files/59466/59466-0.txt"
+        async with aiohttp.ClientSession() as session:
+            async with session.get(uri) as data:
+                text_data = await data.text()
```
Comment (lmazuel) [inline sdk/storage/azure-storage-blob/tests/test_common_blob_async.py]: I meant use async iterator:
https://aiohttp.readthedocs.io/en/stable/streams.html#asynchronous-iteration-support

Comment (lmazuel) [reply sdk/storage/azure-storage-blob/tests/test_common_blob_async.py]: iter_chunked

Comment (rakshith91) [reply sdk/storage/azure-storage-blob/tests/test_common_blob_async.py]: fixed

**Pull request merged**